improver_tests/nbhood/recursive_filter/test_RecursiveFilter.py

```diff
@@ -521,6 +521,65 @@ def test_coordinate_reordering_with_different_smoothing_coefficients(self):
         )
         self.assertArrayAlmostEqual(result.data[0], expected_result)
 
+    def test_mask_zeros(self):
+        """Test that any zeros in the data are the same at the beginning and
+        end of the recursive filtering with the mask_zeros option."""
+        zeros = self.cube.data == 0.0
+        plugin = RecursiveFilter(
+            iterations=self.iterations,
+        )
+        result = plugin(
+            self.cube,
+            smoothing_coefficients=self.smoothing_coefficients,
+            mask_zeros=True,
+        )
+        result_zeros = result.data == 0.0
+        self.assertArrayEqual(zeros, result_zeros)
+
+    def test_mask_zeros_same_mask(self):
+        """Test that if mask_zeros is applied to a cube that already
+        has a mask, the end result has the same mask."""
+        mask = np.zeros(self.cube.data.shape)
+        mask[0, 3, :] = 1
+        self.cube.data = np.ma.MaskedArray(self.cube.data, mask=mask)
+        mask_of_cube = np.ma.getmaskarray(self.cube.data)
+        plugin = RecursiveFilter(
+            iterations=self.iterations,
+        )
+        result = plugin(
+            self.cube,
+            smoothing_coefficients=self.smoothing_coefficients,
+            mask_zeros=True,
+        )
+        mask_of_result = np.ma.getmaskarray(result.data)
+        self.assertArrayEqual(mask_of_cube, mask_of_result)
+
+    def test_mask_zeros_result(self):
+        """Test that if the mask_zeros option is on with masked data it
```

Comment:
Can we change this a bit to agree with the language above, like "Test that if mask_zeros is applied to a cube with masked data, it returns the correct result."
I wasn't too sure what it meant y "option is on".

Reply:
I've changed this doc-string to 

>      """Test that if mask_zeros is applied to a cube that contains zeros, the resulting data is correct."""

which changes the meaning slightly. I don't think that the input cube to this test does contain masked data, so I think that this is clearer.